### ISSUE TITLE
[docs] consider switching off of VuePress

### ISSUE THREAD

**Issue by seagreen** [dxos/dxos#6569]
VuePress is deprecated, but that's not a big deal. Honestly it could be an asset if looked at from a certain way.

However, we're hitting a number of issues with it:

### Animations are too long

#6548

### The sidebar mixes up page hierarchy with headers/intra-page ToC

I keep getting confused between this (a huge chunk of hierarchy with lots of pages):
![image](https://github.com/dxos/dxos/assets/1310143/af452bef-6dd5-4b4d-86b5-d57bbbb216da)

with this (which is just a single page with some sections):
![image](https://github.com/dxos/dxos/assets/1310143/1a437149-05b3-4635-b71b-3775e0ea4ec1)

if I'm getting confused then casual passersby are definitely getting confused.

(VitePress for instance shows headers/intra-page ToC [on the right](https://vitepress.dev/guide/what-is-vitepress) instead of the sidebar.)

### The sidebar jerks around during normal operation

Almost each click in the sidebar causes the sidebar contents to jerk upwards, since VuePress only allows one sidebar hierarchy open at once, and you'll almost always be traveling from top-to-bottom and have something open already.

### We'd like another theme anyway

Discussed with @thure, though we don't have an issue for it yet.

**Comment by seagreen** [comment]
I think turning off [nested header links](https://vuepress-theme-hope.github.io/v1/guide/layout/sidebar.html#nested-header-links) and then making the sidebar not collapsible might mitigate multiple of these problems at once, I'll try it out at some point.

**Comment by wittjosiah** [comment]
@seagreen I don't think we have any particular attachment to VuePress. At the time we adopted it, it was just the only thing we found that satisfied our requirements (e.g. vitepress did not support the types of customization that we felt were needed). Previously we were using docusaurus but the webpack build times were killing us.

Astro looks like a solid option today and might be worth trying out at some point for docs, but as you say it's probably not a high priority.

**Comment by seagreen** [comment]
> Previously we were using docusaurus but the webpack build times were killing us.

Wow, well you did your due diligence. Docusaurus was one of the options I was going to mention, alongside VitePress and Astro [Starlight](https://starlight.astro.build/).

> as you say it's probably not a high priority.

Agreed, it works well enough.

**Comment by seagreen** [comment]
I'm not sure who to credit because I didn't change anything, but intra-page sections now show on the right, enormous improvement:

![image](https://github.com/dxos/dxos/assets/1310143/104ee3b1-549e-4c0c-b7c1-324f435806d5)

**Comment by seagreen** [comment]
Oh, it's just screen width. With a wide enough screen you see what's in the screenshot above and it works nicely.
